src/cargo/ops/cargo_add/mod.rs

```diff
@@ -449,6 +452,45 @@ fn resolve_dependency(
     Ok(dependency)
 }
 
+fn fuzzy_lookup(
+    dependency: &mut Dependency,
+    lookup: impl Fn(&str) -> CargoResult<Option<Dependency>>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<Dependency>> {
+    if let Some(rename) = dependency.rename() {
+        // If it has a rename only lookup that name - analogue to Dependency::toml_key()
```

Comment:
This comment is a re-phrasing of the code and isn't describing the motivation behind it.

Oh, I think I see the key reason for this. Before we did lookups with `toml_key`.  This manually implements `toml_key` so that fuzzy lookups only happen on package names, not dependency names, to scope this to only happen when other normalization happens (`PackageRegistr::query_vec`).  Is that it?

Reply:
The intention is to mirror the behavior of `Dependency::toml_key()` to lookup _only_ on the rename if that is present. I agree with your description that fuzzy lookups should only happen on packages names.

I will write `Mirror behavior of to Dependency::toml_key()`

Reply:
That describes the whole function in terms of checking `rename` and then `name`.

The specific nuance is why you are doing that, rather than using `toml_key`.

Something like
```
Manually implement `toml_key` to restrict fuzzy lookups to only package names to mirror `PackageRegistry::query_vec`
```

Reply:
I used the suggested comment.